src/libs/ReportUtils.ts

```diff
@@ -1563,6 +1563,14 @@ function isOneTransactionReport(reportID: string): boolean {
     return ReportActionsUtils.getOneTransactionThreadReportID(reportID, reportActions) !== null;
 }
 
+function isPayAtEndExpenseReport(reportID: string): boolean {
+    if (!isOneTransactionReport(reportID)) {
+        return false;
+    }
+
+    return TransactionUtils.isPayAtEndExpense(TransactionUtils.getAllReportTransactions(reportID)[0]);
```

Comment:
Can we just `useOnyx` in the component to get this transaction instead of using `TransactionUtils.getAllReportTransactions`? That way the component will re-render whenever the transaction updates.

Reply:
It actually doesn't. The `Onyx.connect` you see here (and in other libs) just runs the `callback` when the value changes and has no effect on any React components. 

So that can cause subtle issues where a component initially renders correctly using functions like this one, but then it doesn't re-render when relevant Onyx values change because it isn't subscribed to them. To subscribe we need to use `useOnyx` or `withOnyx`.

Reply:
Thanks for the explanation!

I updated it to use `useOnyx` where it was possible, but in the `canIOUBePaid` call we need to pass down the transactions.

Also the transactions might be undefined because [the `approveRequest` call](https://github.com/Expensify/App/blob/b110056916c6977bf44c97d4cddcfc8d9849c39f/src/libs/actions/IOU.ts#L6865) can safely get the latest transactions I think, since it happens on user click.

Let me know what you think